### PR TITLE
Add fr.henri2h.minestrix

### PR DESCRIPTION
**Pull request by henri2h**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(Additional pulseaudio, network and notifications)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link: https://gitlab.com/minestrix/minestrix-flutter**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by henri2h** [comment]
bot, build fr.henri2h.minestrix

**Comment by henri2h** [comment]
bot, build fr.henri2h.minestrix

**Comment by henri2h** [comment]
bot, build fr.henri2h.minestrix

File: fr.henri2h.minestrix.desktop [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Type=Application
+Name=MinesTRIX
+Icon=fr.henri2h.minestrix
+Exec=env GDK_GL=gles minestrix %U
```
Comment (hfiguiere) [inline fr.henri2h.minestrix.desktop:5]: this is likely incorrect.

File: fr.henri2h.minestrix.json [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    "app-id": "fr.henri2h.minestrix",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "43",
+    "sdk": "org.gnome.Sdk",
+    "command": "minestrix",
+    "separate-locales": false,
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=fallback-x11",
```
Comment (hfiguiere) [inline fr.henri2h.minestrix.json:11]: these two are mutually exclusive.

File: fr.henri2h.minestrix.json [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        "--socket=wayland",
```
Comment (hfiguiere) [inline fr.henri2h.minestrix.json:12]: does it support Wayland?

File: fr.henri2h.minestrix.json [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        "--socket=pulseaudio",
+        "--share=network",
+        "--device=all",
+        "--filesystem=xdg-download",
+        "--talk-name=org.gnome.Notifications"
+    ],
+    "build-options": {
+        "build-args": [
+            "--share=network"
```
Comment (hfiguiere) [inline fr.henri2h.minestrix.json:21]: build args are not allowed. You need to make sure the build is offline.

File: fr.henri2h.minestrix.json [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+                {
+                    "type": "archive",
+                    "url": "https://github.com/open-source-parsers/jsoncpp/archive/refs/tags/1.7.5.tar.gz",
+                    "sha256": "4338c6cab8af8dee6cdfd54e6218bd0533785f552c6162bb083f8dd28bf8fbbe"
+                }
+            ]
+        },
+        {
+            "name": "olm",
+            "buildsystem": "cmake",
```
Comment (hfiguiere) [inline fr.henri2h.minestrix.json:63]:
```suggestion
            "buildsystem": "cmake-ninja",
```

File: fr.henri2h.minestrix.json [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://github.com/open-source-parsers/jsoncpp/archive/refs/tags/1.9.5.tar.gz",
+                    "sha256": "f409856e5920c18d0c2fb85276e24ee607d2a09b5e7d5f0a371368903c275da2"
+                }
+            ]
+        },
+        {
+            "name": "libjsoncpp",
```
Comment (hfiguiere) [inline fr.henri2h.minestrix.json:44]: why two diffrerent versions?

File: fr.henri2h.minestrix.json [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+            "buildsystem": "cmake",
```
Comment (hfiguiere) [inline fr.henri2h.minestrix.json:45]:
```suggestion
            "buildsystem": "cmake-ninja",
```

File: fr.henri2h.minestrix.json [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+        },
+        {
+            "name": "minestrix",
+            "buildsystem": "simple",
+            "only-arches": [
+                "x86_64",
+                "aarch64"
+            ],
+            "build-commands": [
+                "ls minestrix",
```
Comment (hfiguiere) [inline fr.henri2h.minestrix.json:95]:
```suggestion
```

File: fr.henri2h.minestrix.json [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+                }
+            ]
+        },
+        {
+            "name": "minestrix",
+            "buildsystem": "simple",
+            "only-arches": [
+                "x86_64",
+                "aarch64"
+            ],
```
Comment (hfiguiere) [inline fr.henri2h.minestrix.json:93]:
```suggestion
```

File: flathub.json [hunk 10]
```diff
@@ -0,0 +1,6 @@
+{
+  "only-arches": [
+    "x86_64"
```
Comment (hfiguiere) [inline flathub.json:3]: this is disappointing.

File: fr.henri2h.minestrix.json (hunk 4 above)
Comment (henri2h) [reply fr.henri2h.minestrix.json]: done

File: fr.henri2h.minestrix.desktop (hunk 1 above)
Comment (henri2h) [reply fr.henri2h.minestrix.desktop]: I removed the env GDK_GL=gles which didn't made any sense indeed

File: flathub.json (hunk 10 above)
Comment (henri2h) [reply flathub.json:3]: Well I can have a look to add aarch64 build in the CI

File: fr.henri2h.minestrix.json (hunk 9 above)
Comment (henri2h) [reply fr.henri2h.minestrix.json:69]: Not sure what the suggested change is

**Comment by henri2h** [comment]
bot, build fr.henri2h.minestrix

File: flathub.json (hunk 10 above)
Comment (henri2h) [reply flathub.json:3]: Don't have anything to make arm64 build at the moment.

File: fr.henri2h.minestrix.json (hunk 9 above)
Comment (hfiguiere) [reply fr.henri2h.minestrix.json:69]: to remove it?

File: fr.henri2h.minestrix.json [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+        "--filesystem=xdg-download",
+        "--talk-name=org.gnome.Notifications"
+    ],
+    "modules": [
+        {
+            "name": "libjsoncpp",
+            "only_arches": [
+                "x86_64",
+                "aarch64"
+            ],
```
Comment (hfiguiere) [inline fr.henri2h.minestrix.json:23]:
```suggestion
```

File: fr.henri2h.minestrix.json [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://gitlab.matrix.org/matrix-org/olm.git",
+                    "tag": "3.2.14"
+                }
+            ]
+        },
+        {
+            "name": "libsecret",
```
Comment (hfiguiere) [inline fr.henri2h.minestrix.json:49]: it is in the runtime, why do yo need to build it?

Comment (henri2h) [reply fr.henri2h.minestrix.json:49]: ok, didn't new, going to remove it.

**Comment by henri2h** [comment]
bot, build fr.henri2h.minestrix

**Pull request closed without merging**